metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -1205,31 +1194,46 @@ def get_tables_for_dataset(
                             with_data_read_permission=self.config.profiling.enabled,
                         )
                     )
+                    table_items.clear()
+
+            # Sharded tables don't have partition keys, so it is safe to add to the list as
+            # it should not affect the number of tables will be touched in the partitions system view.
+            # Because we have the batched query of get_tables_for_dataset to makes sure
+            # we won't hit too many tables queried with partitions system view.
+            # The key in the map is the actual underlying table name and not the friendly name and
+            # that's why we need to get the actual table names and not the normalized ones.
+            table_items.update(
+                {value.table_id: value for value in sharded_tables.values()}
+            )
 
-                self.db_tables[dataset_name] = bigquery_tables
-
-                self.report.metadata_extraction_sec[
-                    f"{project_id}.{dataset_name}"
-                ] = round(timer.elapsed_seconds(), 2)
+            if table_items:
+                bigquery_tables.extend(
+                    BigQueryDataDictionary.get_tables_for_dataset(
+                        conn,
+                        project_id,
+                        dataset_name,
+                        table_items,
+                        with_data_read_permission=self.config.profiling.enabled,
+                    )
+                )
```

Comment:
so having read through this section of the code I can say I have little idea what the implications of making these changes will be